webapi/src/test/scala/org/knora/webapi/responders/v2/ontology/OntologyHelpersSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.responders.v2.ontology
+
+import dsp.schema.domain._
+import org.knora.webapi.CoreSpec
+import org.knora.webapi.InternalSchema
+import org.knora.webapi.messages.IriConversions._
+import org.knora.webapi.messages.SmartIri
+import org.knora.webapi.messages.StringFormatter
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+import org.knora.webapi.messages.v2.responder.ontologymessages.OwlCardinality.KnoraCardinalityInfo
+import org.knora.webapi.responders.v2.ontology.Cache._
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+/**
+ * This spec is used to test [[org.knora.webapi.responders.v2.ontology.OntologyHelpers]].
+ */
+class OntologyHelpersSpec extends CoreSpec {
+
+  private implicit val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+  private implicit val timeout                          = 10.seconds
+  private val freeTestOntology = RdfDataObject(
+    path = "test_data/ontologies/freetest-onto.ttl",
+    name = "http://www.knora.org/ontology/0001/freetest"
+  )
+  override lazy val rdfDataObjects: List[RdfDataObject] = List(freeTestOntology)
+
+  val freetestOntologyIri: SmartIri =
+    "http://0.0.0.0:3333/ontology/0001/freetest/v2".toSmartIri.toOntologySchema(InternalSchema)
+
+  "The ontology helper" should {
+
+    "determine the least strict cardinality allowed for a (inherited) property from a list of classes" in {
+
+      val cacheDataFuture: Future[OntologyCacheData] = Cache.getCacheData
+      val cacheData: OntologyCacheData               = Await.result(cacheDataFuture, 2.seconds)
+
+      // define the classes
+      val mayHaveMany  = freetestOntologyIri.makeEntityIri("PubMayHaveMany").toOntologySchema(InternalSchema)  // 0-n
+      val mustHaveSome = freetestOntologyIri.makeEntityIri("PubMustHaveSome").toOntologySchema(InternalSchema) // 1-n
+      val mayHaveOne   = freetestOntologyIri.makeEntityIri("PubMayHaveOne").toOntologySchema(InternalSchema)   // 0-1
+      val mustHaveOne  = freetestOntologyIri.makeEntityIri("PubMustHaveOne").toOntologySchema(InternalSchema)  // 1
+
+      val hasPublicationDateProperty = freetestOntologyIri.makeEntityIri("hasPublicationDate")
+
+      // define all test cases and the expected results (see comments)
+      val testCases: List[Set[SmartIri]] = List(
```

Comment:
To make it even easier to read/check, you could merge `testCases` and `expectedResults` as tuples per test case:

```
val testCases: List(
  (Set(mayHaveMany), Some(MayHaveMany))
)
```
then do something like this:
```
testCases map { (testCase, expected) =>
  assert(getStrictest(testCase) == expected)
}


def getStrictest(set: Set[SmartIri]) = OntologyHelpers.getStrictestCardinalitiesFromClasses(
          set,
          cacheData
        ).get(hasPublicationDateProperty) match {
            case None       => None
            case Some(card) => Some(card.cardinality)
          }
```